### PR TITLE
rpc/client: use compile time assertions instead of methods

### PR DESCRIPTION
**Pull request by odeke-em**

### CODE REVIEW

**Comment by odeke-em** [comment]
@ethanfrey please take a look.

File: rpc/client/httpclient.go [hunk 1]
```diff
@@ -39,17 +39,14 @@ func NewHTTP(remote, wsEndpoint string) *HTTP {
...
-
-func (c *HTTP) _assertIsNetworkClient() NetworkClient {
-	return c
-}
-
-func (c *HTTP) _assertIsEventSwitch() types.EventSwitch {
-	return c
-}
+var (
+	// Some compile time assertions to ensure that
```
Comment (melekes) [inline rpc/client/httpclient.go]: don't think this comment is necessary

File: rpc/client/localclient.go [hunk 2]
```diff
@@ -41,13 +41,12 @@ func NewLocal(node *nm.Node) Local {
 	}
 }
 
-func (c Local) _assertIsClient() Client {
-	return c
-}
+var (
+	aLocal = Local{}
```
Comment (ebuchman) [inline rpc/client/localclient.go]: why this?

**Comment by ebuchman** [comment]
Otherwise lgtm

File: rpc/client/localclient.go (hunk 2 above)
Comment (odeke-em) [reply rpc/client/localclient.go]: Because below, we'd end up creating more than one `Local{}`

File: rpc/client/httpclient.go (hunk 1 above)
Comment (odeke-em) [reply rpc/client/httpclient.go]: Roger that, @melekes I'll update that, thanks.

File: rpc/client/localclient.go (hunk 2 above)
Comment (odeke-em) [reply rpc/client/localclient.go]: However, I'll remove it for clarity and instead use `Local{}` all the way.

**Comment by odeke-em** [comment]
I've updated the PR to address both y'all comments, PTAL.

**Pull request merged**